### DISCUSSION TITLE
Difference between Build for Inferencing and Build for training?

### DISCUSSION THREAD

**Discussion by lamikr** [microsoft/onnxruntime#20820] [General]
I have build the onnxruntime 17.3 now for AMD rocm by using the documentation how to do the build for inferencing and training.
https://onnxruntime.ai/docs/build/inferencing.html
https://onnxruntime.ai/docs/build/training.html

Documentation does not however explain what is the main difference between these builds. If I do the build for training, does it also include the inferencing features? Basically it training build just adds the "--enable_training" in top of the inferencing build.

Then there is also 2 AMD specific builds described in
https://onnxruntime.ai/docs/build/eps.html

and it's little unclear can I also enable both the "-use-rocm --rocm_home <path>" and 
 "--use_migraphx --migraphx_home<path>" flags on same build or does one flag remove some features from another flag?

**Comment by LaurentBerger** [comment]
I'm not an expert in ONNX and I'm facing same issue with ONNX documentation
About training it's here https://onnxruntime.ai/training and https://onnxruntime.ai/docs/get-started/training-pytorch.html and https://onnxruntime.ai/docs/api/c/training_c_cpp_api.html
To train a new model you need 

> In order to train a model with onnxruntime, the following training artifacts must be generated:
> 
>     The training onnx model
>     The checkpoint file
>     The optimizer onnx model
>     The eval onnx model model (optional)
> 
> These training artifacts can be generated as part of an offline step using the python [utilities](https://github.com/microsoft/onnxruntime/blob/main/orttraining/orttraining/python/training/onnxblock/README.md) made available in the onnxruntime-training python package.

**Comment by carzh** [comment]
**Regarding training build vs inference build:**
Inference build is the default build. Using the `--enable_training` build flag will build for training, which includes the inferencing capabilities with the training capabilities added. In other words, the training build contains both inferencing and training functionality.

**Regarding large model training vs on-device training:**
ONNXRuntime supports two types of training capabilities: large model training, which will accelerate a PyTorch Python training script with minimal code changes, and on-device training, which enables training in other platforms / languages. On-device training requires the extra step of generating 3-4 training artifacts. 

We do need better documentation to make clear what ORT offerings are. Let me know if you have any more questions about training.
